staging/src/k8s.io/apiserver/pkg/server/options/audit.go

```diff
@@ -231,7 +259,16 @@ func (o *AuditLogOptions) getWriter() io.Writer {
 
 func (o *AuditLogOptions) advancedApplyTo(c *server.Config) error {
 	if w := o.getWriter(); w != nil {
-		c.AuditBackend = appendBackend(c.AuditBackend, pluginlog.NewBackend(w, o.Format, auditv1beta1.SchemeGroupVersion))
+		backend, err := pluginlog.NewBackend(w, o.Format, auditv1beta1.SchemeGroupVersion)
+		if err != nil {
+			return fmt.Errorf("initializing audit log: %v", err)
+		}
+
+		if o.Mode == modeBatch {
```

Comment:
Extract to a separate function `func applyBuffered(o *AuditBufferedOptions, audit.Backend) audit.Backend`